Corruption seen with compressed IOSurfaces at some sizes

smfr opened the pull request:
#### 50ee1b15d6128f2236c4c0459f8e423260708486
<pre>
Corruption seen with compressed IOSurfaces at some sizes
<a href="https://bugs.webkit.org/show_bug.cgi?id=296781">https://bugs.webkit.org/show_bug.cgi?id=296781</a>
<a href="https://rdar.apple.com/157238335">rdar://157238335</a>

Reviewed by Matt Woodrow and Tim Horton.

For small layer backing stores, we use a software renderer into an IOSurface (via
ImageBufferShareableMappedIOSurfaceBitmapBackend), which is not a valid configuration
when using lossless IOSurface compression. CGBitmapContextCreate() would fail because
IOSurfaceGetBytesPerRow() returns an incompatible value.

Add a RELEASE_LOG_ERROR when this happens, and also fix the existing logging for
IOSurfaceSetOwnershipIdentity() failure to log the surfaceID, since that&apos;s what
other tools use.

Fix by passing UseLosslessCompression::No in `ImageBufferShareableMappedIOSurfaceBitmapBackend::create()`.

* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::createBitmapPlatformContext):
(WebCore::IOSurface::setOwnershipIdentity):
* Source/WebKit/WebProcess/GPU/graphics/cocoa/ImageBufferShareableMappedIOSurfaceBitmapBackend.cpp:
(WebKit::ImageBufferShareableMappedIOSurfaceBitmapBackend::create):

Canonical link: <a href="https://commits.webkit.org/298108@main">https://commits.webkit.org/298108@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/efaae971d077570f67e2c8d8b4a4d0584b310ef6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114294 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34039 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24503 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120458 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/65013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116183 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34670 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42600 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/86854 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/65013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117242 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27610 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102647 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67247 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26791 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20772 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64147 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96984 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20888 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123668 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41308 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30802 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/95682 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41685 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98847 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95465 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24326 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40609 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18444 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/37367 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41188 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/46696 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40783 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44090 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42531 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fd02acf5f592780a095dd7dd304e1a095e220396. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48806

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/efaae971d077570f67e2c8d8b4a4d0584b310ef6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48806

smfr committed:
Corruption seen with compressed IOSurfaces at some sizes
https://bugs.webkit.org/show_bug.cgi?id=296781
rdar://157238335

Reviewed by Matt Woodrow and Tim Horton.

For small layer backing stores, we use a software renderer into an IOSurface (via
ImageBufferShareableMappedIOSurfaceBitmapBackend), which is not a valid configuration
when using lossless IOSurface compression. CGBitmapContextCreate() would fail because
IOSurfaceGetBytesPerRow() returns an incompatible value.

Add a RELEASE_LOG_ERROR when this happens, and also fix the existing logging for
IOSurfaceSetOwnershipIdentity() failure to log the surfaceID, since that's what
other tools use.

Fix by passing UseLosslessCompression::No in `ImageBufferShareableMappedIOSurfaceBitmapBackend::create()`.

* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::createBitmapPlatformContext):
(WebCore::IOSurface::setOwnershipIdentity):
* Source/WebKit/WebProcess/GPU/graphics/cocoa/ImageBufferShareableMappedIOSurfaceBitmapBackend.cpp:
(WebKit::ImageBufferShareableMappedIOSurfaceBitmapBackend::create):

Canonical link: https://commits.webkit.org/298108@main
